### PR TITLE
Improve LocalMethodCallFinder performance

### PR DESCRIPTION
**Pull request by staabm**
leads to a 86% faster analysis when running rector on a folder of the phpstan-src codebase

grafik

### CODE REVIEW

File: src/PhpParser/NodeFinder/LocalMethodCallFinder.php [hunk 1]
```diff
@@ -44,16 +44,16 @@ public function match(ClassMethod $classMethod): array
         $matchingMethodCalls = [];
 
         foreach ($methodCalls as $methodCall) {
-            $callerType = $this->nodeTypeResolver->getType($methodCall->var);
-            if (! $callerType instanceof TypeWithClassName) {
+            if (! $this->nodeNameResolver->isName($methodCall->name, $classMethodName)) {
```
Comment (staabm) [inline src/PhpParser/NodeFinder/LocalMethodCallFinder.php:47]: doing the cheaper check first so we don't need to resolve the type of every `methodcall->var`

**Comment by staabm** [comment]
the failling job is unrelated. I saw this very same error also on other PRs

**Comment by samsonasik** [comment]
That probably due to new phpdoc-parser release https://github.com/phpstan/phpdoc-parser/releases/tag/1.20.1

**Review by samsonasik — APPROVED** [review]
Looks good, thank you @staabm

**Pull request merged**